### PR TITLE
Can Crusher

### PR DESCRIPTION
**Pull request by DeityLink**
https://cdn.discordapp.com/attachments/246096207523348481/594974501570084914/dreamseeker_2019-06-30_21-31-16-86_with_sound.webm
crushingcans

:cl:
* rscadd: Clicking on an empty soda can with HARM intent on will crush the can in your hand, making a satisfying sound. Crushed cans are considered a trash item so they will persist between rounds. They can no longer be cut with wirecutters or used to hold reagents, however they can still be ground into aluminum using a mortar, or fed to an auto-lathe for a bit of iron.
* rscadd: Crushed cans deal 2 brute damage instead of one when thrown, and can be thrown slightly further and faster that regular cans.

### CODE REVIEW

**Comment by DeityLink** [comment]
The sprite is automatically generated and cached for each can.

**Comment by jknpj** [comment]
give them more throwing damage

**Comment by MadmanMartian** [comment]
A crushed can should be a smaller class

**Comment by Irradiation** [comment]
*sip
They don't make these cans like they used to.

**Comment by Kharkov72** [comment]
*sip
Groans soda, now THAT'S a drink

**Comment by DeityLink** [comment]
@jknpj I shall allow it

@MadmanMartian they're already of W_CLASS_TINY like most reagent containers/food/drinks

**Comment by ghost** [comment]
https://www.youtube.com/watch?v=XtOjrGCePOU

**Comment by DeityLink** [comment]
~I think I'll just have the crushed cans become a trash item, which will allow for it being persistent and all that jazz.~ Ok nevermind this is way too much work for what it's worth. The easiest way would be to created a trash type and icon for every single cans so fuck that. Cans won't be trash, unless there's a way to save the appearance/icon or something beyond its icon_state.

TL;DR: there's a way but it's too much work for what this feature is worth.

**Comment by Kurfursten** [comment]
image

This was definitely a good use of the 20 minutes I had to spend before leaving for dinner

**Comment by DeityLink** [comment]
It's supposed to zoom on his face in the last 3 panels. You get a B

**Comment by Lokiamis** [comment]
Do they crush if you hit someone with them?

**Comment by DeityLink** [comment]
No, you have to crush it in your hand. Maybe later we can have it crush if you step on it.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
wait
can I still use a crushed can as a perfectly functioning reagent container
wew if so

**Comment by DeityLink** [comment]
shhhhh

**Comment by Blithering** [comment]
Drunk people should be able to attack themselves on the head with it for extreme can crushing.

**Comment by MadmanMartian** [comment]
Why doesn't this just make a new object, rather than deleting half of the variables?

This would be better off as a trash/ subtype

**Comment by SonixApache** [comment]
do they still recycle (read: mortar grind) into aluminium?

**Comment by DeityLink** [comment]
The icon being procedurally generated means we can't save its icon and have it be persistent trash. @MadmanMartian . I gave a shot at having it be a trash subtype but unless we make a subtype for every kind of cans (and I'm not doing that) it won't look as good.

@SonixApache yep

**Comment by SonixApache** [comment]
Why would you have to make a subtype for every type of can? Can't you just pass the icon state off the can you're crushing to the new trash?
```
var/obj/item/trash/can/maketrash = new /obj/item/trash/can
maketrash.iconstate = iconstate
maketrash.forcemove(usr.loc)
qdel(src)
```
or something

\*edit*
oh, PERSISTENT trash. Right. Yeah, that's a whole different can of worms.

**Comment by DeityLink** [comment]
Actually there might be a way, I'll think about it.

**Comment by DeityLink** [comment]
Alright I've moved the procedural icon generation to the trash item's New(), and in such a way that it's compatible with persistent trash, so say hello to persistent crushed cans people. @MadmanMartian

**Comment by press-mark** [comment]
can we crush cans on our heads?

**Comment by MadmanMartian** [comment]
I was asking moreso that you avoid unexpected consequences from having it be a heavily modified item, rather than just making a different item

**Comment by Kurfursten** [comment]
conflicts

File: code/modules/reagents/reagent_containers/food/drinks.dm [hunk 1]
```diff
@@ -619,15 +619,25 @@
...
 		return
-	return ..()
+	if (reagents.total_volume > 0)
+		return ..()
+	else if (user.a_intent == I_HURT)
+		var/turf/T = get_turf(user)
+		user.drop_item(src, T, 1)
+		var/obj/item/trash/soda_cans/crushed_can = new (T, icon_state = icon_state)
+		crushed_can.name = "crushed [name]"
+		crushed_can.attack_hand(user)
```
Comment (MadmanMartian) [inline code/modules/reagents/reagent_containers/food/drinks.dm]: Why is this being used?

File: code/game/objects/items/trash.dm [hunk 2]
```diff
@@ -193,4 +193,40 @@
...
+	icon = 'icons/obj/drinks.dmi'
+	icon_state = "crushed_can"
+	starting_materials = list(MAT_IRON = 50)
+	w_type=RECYK_METAL
+	force = 0
+	throwforce = 2
+	throw_range = 8
+	throw_speed = 3
+
+/obj/item/trash/New(var/loc, var/age, var/icon_state, var/color, var/dir, var/pixel_x, var/pixel_y)
```
Comment (MadmanMartian) [inline code/game/objects/items/trash.dm]: Shouldn't this be /obj/item/trash/soda_cans?

**Comment by MadmanMartian** [comment]
What about hitting the can with something blunt, like a toolbox or a hammer?

**Comment by MadmanMartian** [comment]
Also, currently the contents of the can are not taken into account. Have them splash onto the floor or over the user.

**Comment by DeityLink** [comment]
Fixed your isses @MadmanMartian 

Reguarding your last two points 

>What about hitting the can with something blunt, like a toolbox or a hammer?

that'll be for another PR. Others suggested using your head, that should have a chance to give you brain damage I guess.

>Also, currently the contents of the can are not taken into account. Have them splash onto the floor or over the user.

You can only crush an empty can for now.

**Comment by Intigracy** [comment]
crush a can for jay owen
